cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -388,19 +388,10 @@ func (g *gitHubRepository) downloadFilesFromRelease(release *github.RepositoryRe
 			return false, nil
 		}
 		if redirect != "" {
-			req, err := http.NewRequestWithContext(ctx, http.MethodGet, redirect, http.NoBody)
-			if err != nil {
-				retryError = errors.Wrapf(err, "failed to download file %q from %q release via redirect location %q: failed to create request", *release.TagName, fileName, redirect)
-				return false, nil
-			}
-
-			response, err := http.DefaultClient.Do(req) //nolint:bodyclose // (NB: The reader is actually closed in a defer)
-			if err != nil {
-				retryError = errors.Wrapf(err, "failed to download file %q from %q release via redirect location %q", *release.TagName, fileName, redirect)
-				return false, nil
-			}
-			reader = response.Body
+			retryError = errors.New("unexpected redirect: http.DefaultClient should be able to handle redirects in DownloadReleaseAsset")
+			return true, nil
```

Comment:
Could also be 
```suggestion
			return false, errors.New("unexpected redirect: http.DefaultClient should be able to handle redirects in DownloadReleaseAsset")
```

but would then require to handle the error in L378.

Reply:
I think the question here is if we want to retry when we hit this line.

Returning an error here immediately "stops" PollImmediate

Reply:
We could follow the pattern we have above:
```go
			retryError = errors.New("unexpected redirect: http.DefaultClient should be able to handle redirects in DownloadReleaseAsset")
			return false, retryError
```

Reply:
Thx!